orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RecreateUpdateStrategy.groovy

```diff
@@ -0,0 +1,220 @@
+import com.netflix.spinnaker.orca.pipeline.model.SyntheticStageOwner
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.ApplicationContext
+import org.springframework.context.ApplicationContextAware
+import org.springframework.stereotype.Component
+
+import static com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder.newStage
+
+@Slf4j
+@Component
+class RecreateUpdateStrategy implements Strategy, ApplicationContextAware {
+  final String name = "recreateupdate"
+  final static int DEFAULT_TERMINATION_WAIT_TIME_IN_SECONDS = 30
+
+  @Autowired
+  ResizeServerGroupStage resizeServerGroupStage
+
+  @Autowired
+  WaitStage waitStage
+
+  @Autowired
+  PipelineStage pipelineStage
+
+  @Autowired
+  DetermineTargetServerGroupStage determineTargetServerGroupStage
+
+  @Autowired
+  TargetServerGroupResolver targetServerGroupResolver
+
+  @Override
+  <T extends Execution<T>> List<Stage<T>> composeFlow(Stage<T> stage) {
+    def stages = []
+    def stageData = stage.mapTo(RecreateUpdateStrategyData)
+    def cleanupConfig = AbstractDeployStrategyStage.CleanupConfig.fromStage(stage)
+
+    Map baseContext = [
+      (cleanupConfig.location.singularType()): cleanupConfig.location.value,
+      cluster                                : cleanupConfig.cluster,
+      credentials                            : cleanupConfig.account,
+      cloudProvider                          : cleanupConfig.cloudProvider,
+    ]
+
+    if (!stageData.delayBeforeCleanup) {
+      stageData.delayBeforeCleanup = DEFAULT_TERMINATION_WAIT_TIME_IN_SECONDS
+    }
+
+    stage.context.capacity = [
+      min    : 0,
+      max    : 0,
+      desired: 0
+    ]
+
+
+    def targetPercentages = stageData.getTargetPercentages()
+    if (targetPercentages.size() == 0 || targetPercentages[-1] != 100) {
+      log.info("Inserting implicit 100% final target percentage...")
+      targetPercentages.add(100)
+    }
+
+    def findContext = baseContext + [
+      target        : TargetServerGroup.Params.Target.current_asg_dynamic,
+      targetLocation: cleanupConfig.location,
+    ]
+
+    stages << newStage(
+      stage.execution,
+      determineTargetServerGroupStage.type,
+      "Determine Deployed Server Group",
+      findContext,
+      stage,
+      SyntheticStageOwner.STAGE_BEFORE
+    )
+
+    targetPercentages.forEach({ p ->
+      def source
+      try {
+        source = getSource(targetServerGroupResolver, stageData, baseContext)
+      } catch (Exception e) {
+        stages = []
+        return stages
+      }
+      def resizeContext = baseContext + [
+        target        : TargetServerGroup.Params.Target.current_asg_dynamic,
+        action        : ResizeStrategy.ResizeAction.scale_exact,
+        source        : source,
+        targetLocation: cleanupConfig.location,
+        scalePct      : p,
+        pinCapacity   : p < 100, // if p = 100, capacity should be unpinned
```

Comment:
Do we need all this resize/scaling logic if we're just recreating?

Reply:
I can remove the scaling because I pass the exactly size.